### DISCUSSION TITLE
python cell prints graphviz code text which is rendered

### DISCUSSION THREAD

**Discussion by drdozer** [quarto-dev/quarto-cli#6508] [Q&A] [answered]
### Description

I have a python code block that when it runs prints out a graphviz string. Is there a way to get that output treated as a grpahviz code-block, which then gets rendered like any other code block?

**Comment by cscheid** [comment, marked as answer]
(I haven't tested this, but it should work.)

It's an advanced hack, but yes. You want to use "output: asis" in the Python code cell metadata, and then produce output that corresponds exactly to the dot input you'd use here: https://quarto.org/docs/authoring/diagrams.html.

This should work because our `dot` handlers are run after the engines execute, and so we should be able to see your `dot` input.

**Reply by drdozer** [reply]
Thanks - I'm trying various versions of this, but not having much joy. I have tried printing and I have tried returning the string. It seems to just dump the string in rather than treating it as a codeblock in turn.

```{python}
#| output: asis

backtick = "`"
t3 = backtick * 3

f"""
{t3}{{mermaid}}
graph TD:
  A --> B;
{t3}
"""
```

**Reply by drdozer** [reply]
So it does work if I print the markdown. It renders correctly in the html rendered version of the document, but does not render in the visual studio plugin. Thanks for your help.

**Reply by cscheid** [reply]
> but does not render in the visual studio plugin

I believe that; there's no way for the VS Code plugin to know that the specific code you're generating is going to cause diagrams to be rendered.

1. that's the nature of `output: asis` tricks. 
2. The VS Code plugin repo is https://github.com/quarto-dev/quarto, and that would be a new feature request to be discussed/made there.
